pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_runtime_reconcile.go

```diff
@@ -79,7 +79,10 @@ func (c *runtimeReconcile) reconcile(ctx context.Context, cm *operatorapiv1.Clus
 			enabledRegistrationDrivers = append(enabledRegistrationDrivers, registrationDriver.AuthType)
 			if registrationDriver.AuthType == "awsirsa" {
 				config.HubClusterArn = registrationDriver.HubClusterArn
-				config.AutoApprovedARNPatterns = registrationDriver.AutoApprovedIdentities
+				var AutoApprovedIdentities = registrationDriver.AutoApprovedIdentities
```

Comment:
I do not think this line is necessary.

can we just strings.join(registrationDriver.AutoApprovedIdentities, ",")

and why CSR part does not need this change?

Reply:
Addressed. I think we overlooked it. Fixed for CSR as well, as well as added an integration test for csr.